data-prepper-plugins/kafka-source/src/main/java/org/opensearch/dataprepper/plugins/source/KafkaSource.java

```diff
@@ -0,0 +1,107 @@
+package org.opensearch.dataprepper.plugins.source;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.source.Source;
+import org.opensearch.dataprepper.plugins.consumer.MultithreadedConsumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@SuppressWarnings("deprecation")
+@DataPrepperPlugin(name = "kafka", pluginType = Source.class, pluginConfigurationType = KafkaSourceConfig.class)
+public class KafkaSource implements Source<Record<Event>> {
+
+	private static final Logger logger = LoggerFactory.getLogger(KafkaSource.class);
+	private final KafkaSourceConfig sourceConfig;
+	final List<MultithreadedConsumer> consumers = new ArrayList<>();
+	private ExecutorService executorService;
+
+	@DataPrepperPluginConstructor
+	public KafkaSource(final KafkaSourceConfig sourceConfig) {
+		this.sourceConfig = sourceConfig;
+	}
+
+	@Override
+	public void start(Buffer<Record<Event>> buffer) {
+		int consumerCount = KafkaSourceConfig.CONSUMER_COUNT;
+		String consumerGroups = KafkaSourceConfig.CONSUMER_GROUP_NAME;
+		Properties props = getConsumerProperties();
+		try {
+			if (executorService == null || executorService.isShutdown()) {
+				executorService = Executors.newFixedThreadPool(consumerCount);
+				IntStream.range(0, consumerCount).forEach(index -> {
+					String consumerId = Integer.toString(index + 1);
+					MultithreadedConsumer kafkaSourceConsumer = new MultithreadedConsumer(consumerId, consumerGroups,
+							props, sourceConfig, buffer);
+					consumers.add(kafkaSourceConsumer);
+					executorService.submit(kafkaSourceConsumer);
+				});
+			}
+		} catch (Exception e) {
+			logger.error("Exception while reading records from the Kafka topic...{}", e.getMessage());
```

Comment:
I don't think this message is quite correct. An exception thrown here would happen during setup.

Also, please use the full stack trace here. This exception could come from a variety of locations and it will be hard to track with only the message.

e.g.

```
logger.log("Failed to setup Kafka source.", e);
```